lib/ts/recipe/session/sessionRecipe.ts

```diff
@@ -162,8 +157,8 @@ export default class SessionRecipe extends RecipeModule {
     getHandshakeInfo = async (): Promise<HandshakeInfo> => {
         if (this.handshakeInfo === undefined) {
             if (this.checkIfInServerlessEnv()) {
-                let handshakeInfo = await getHandshakeInfoFromFileIfExists();
-                if (handshakeInfo !== null) {
+                let handshakeInfo = await getDataFromFileIfExists<HandshakeInfo>(HANDSHAKE_INFO_FILE_PATH);
```

Comment:
Arent you supposed to use JSON.parse here?

Reply:
No, JSON parse is done by the common retrieving function